script.js

```diff
@@ -1,4 +1,193 @@
+    }
+  }
+};
+
+var getPlayerDisplayRoll = function () {
+  var output = `Welcome Player ${currentPlayer}. `;
+
+  var array = getCurrentPlayer(currentPlayer);
+
+  //populate player's dice roll
+  playerDiceRoll(array, NUMBER_OF_DICE_PLAYED);
+
+  output += "<br> You rolled ";
+
+  for (var i = 0; i < array.length; i += 1) {
+    if (i == array.length - 1) {
+      output += ` and `;
+    } else if (i != 0) {
+      output += `, `;
+    }
+    output += `${array[i]} for Dice ${i + 1}`;
+  }
+
+  //VARIABLE NUMBER OF DICE - auto generate number if dice is more than 2
+  if (NUMBER_OF_DICE_PLAYED > 2) {
+    output +=
+      " . <br> Please type AUTO_GENERATE_COMBINED_NUMBER or any key and submit for computer to Auto-Generate Combined Number.";
+  } else {
+    output +=
+      " . <br> Choose the order of the dice or type AUTO_GENERATE_COMBINED_NUMBER for computer to Auto-Generate Combined Number.";
+  }
+  return output;
+};
+
+var autoGenerateCombinedNumber = function () {
+  var final = "";
+  var array = getCurrentPlayer();
+  if (currentMode == MODE_LOWEST_WINS) {
+    //generate lowest number possible from array
+    array.sort((a, b) => a - b);
+  } else {
+    //geenerate highest number possible from array
+    array.sort((a, b) => b - a);
+  }
+  for (var i = 0; i < array.length; i += 1) {
+    final += `${array[i]}`;
+  }
+  if (currentPlayer == 1) {
+    playerOneFinal = final;
+  } else {
+    playerTwoFinal = final;
+  }
+  return final;
+};
+
+var calculateWinner = function () {
+  var output = "";
+  if (
+    (playerOneFinal > playerTwoFinal && currentMode == MODE_NORMAL) ||
+    (playerOneFinal < playerTwoFinal && currentMode == MODE_LOWEST_WINS)
+  ) {
+    output += `Player 1 Wins! Player 1: ${playerOneFinal} vs Player 2: ${playerTwoFinal}`;
+    playerOneScore += 1;
+  } else if (
+    (playerOneFinal < playerTwoFinal && currentMode == MODE_NORMAL) ||
+    (playerOneFinal < playerTwoFinal && currentMode == MODE_LOWEST_WINS)
+  ) {
+    output += `Player 2 Wins! Player 1: ${playerOneFinal} vs Player 2: ${playerTwoFinal}`;
+    playerTwoScore += 1;
+  } else output += `It's a draw`;
+
+  return output;
+};
+
+var getPlayerCombinedResult = function (order) {
+  var total = updatePlayerTotal(order);
+  var output = `Player ${currentPlayer}, you chose Dice ${order} first.
+  <br>Your number is ${total}`;
+  if (currentPlayer == 1) {
+    currentPlayer += 1;
+    currentState = STATE_DICE_ROLL;
+    output += `<br> It is now Player ${currentPlayer}'s turn.`;
+  } else {
+    // get result of battle;
+    output += `<br>` + calculateWinner();
+
+    //SCORE - Adding Leadership Board
+    output += `<br><br> 
+    <h4> Leadership Board </h4>
+    Player 1: ${playerOneScore}
+    <br>
+    Player 2: ${playerTwoScore}`;
+  }
+
+  return output;
+};
+
+var updateNumberOfDiceInGame = function (diceNumber) {
+  if (diceNumber > NUMBER_OF_DICE_PLAYED) {
+    NUMBER_OF_DICE_PLAYED = diceNumber;
+    return diceNumber;
```

Comment:
instead of returning diceNumber, I would remove this line and just have the return statement on line 156.